src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -490,10 +492,12 @@ void DistributedSearchTreeImpl<DeviceType>::sortResults(
                   Kokkos::Impl::min_max_functor<View>(keys), reducer);
   if (result.min_val == result.max_val)
     return;
-  Kokkos::BinSort<View, Comp> bin_sort(
-      keys, Comp(n / 2, result.min_val, result.max_val), true);
-  bin_sort.create_permute_vector();
-  applyPermutations(bin_sort, other_views...);
+
+  View keys_clone(Kokkos::ViewAllocateWithoutInitializing("keys"), keys.size());
+  Kokkos::deep_copy(keys_clone, keys);
+  auto const permutation = ArborX::Details::sortObjects(keys_clone);
```

Comment:
Comment that we could in principle avoid making a copy.

Reply:
I'm not sure what you mean here; or rather how do actually avoid it.

Reply:
We weren't making a copy before and Daniel had commented in the cover of this PR that he measured it does not matter.